test/truefi2/lines-of-credit/DebtToken.test.ts

```diff
@@ -0,0 +1,166 @@
+import { expect, use } from 'chai'
+import { MockProvider, solidity } from 'ethereum-waffle'
+import { BigNumberish, Wallet } from 'ethers'
+
+import { beforeEachWithFixture, parseEth, setupTruefi2 } from 'utils'
+
+import { DebtToken, DebtToken__factory, MockTrueCurrency, MockTrueCurrency__factory } from 'contracts'
+
+use(solidity)
+
+describe('DebtToken', () => {
+  enum DebtTokenStatus {
+    Defaulted = 4,
+    Liquidated
+  }
+
+  const payback = async (wallet: Wallet, amount: BigNumberish) => token.mint(debtToken.address, amount)
+  const debt = parseEth(1)
+
+  let lender: Wallet
+  let borrower: Wallet
+  let safu: Wallet
+  let debtToken: DebtToken
+  let token: MockTrueCurrency
+  let poolAddress: string
+  let provider: MockProvider
+
+  beforeEachWithFixture(async (wallets, _provider) => {
+    [lender, borrower, safu] = wallets
+    provider = _provider
+
+    const { standardPool } = await setupTruefi2(lender, provider)
+    poolAddress = standardPool.address
+    token = MockTrueCurrency__factory.connect(await standardPool.token(), lender)
+    debtToken = await new DebtToken__factory(lender).deploy()
+    await debtToken.initialize(standardPool.address, lender.address, borrower.address, safu.address, debt)
+  })
+
+  describe('Constructor', () => {
+    it('correctly takes token from pool', async () => {
+      expect(await debtToken.token()).to.equal(token.address)
+    })
+
+    it('sets pool address', async () => {
+      expect(await debtToken.pool()).to.equal(poolAddress)
+    })
+
+    it('sets borrowers debt', async () => {
+      expect(await debtToken.debt()).to.equal(debt)
+    })
+
+    it('sets erc20 params', async () => {
+      expect(await debtToken.name()).to.equal('TrueFi Debt Token')
+      expect(await debtToken.symbol()).to.equal('DEBT')
+      expect(await debtToken.decimals()).to.equal(18)
+    })
+
+    it('mints tokens for the lender', async () => {
+      expect(await debtToken.balanceOf(lender.address)).to.equal(debt)
+      expect(await debtToken.totalSupply()).to.equal(debt)
+    })
+
+    it('sets status to Defaulted', async () => {
+      expect(await debtToken.status()).to.equal(DebtTokenStatus.Defaulted)
+    })
+  })
+
+  describe('liquidate', () => {
```

Comment:
We should add back the checks to make sure this reverts if Status < 4. If that's not possible without introducing needless complexity then feel free to close this comment.

Reply:
I don't see a reason to add a check for a state that's not reachable. We can't even test it

Reply:
SGTM, I'll close this comment then.